### ISSUE TITLE
out of memory - at midnight +

### ISSUE THREAD

**Issue by DG12** [influxdata/influxdb#13312]
RuuviCollector inserts  frequently and continuously.
rsyslogd routinely cuts over info and debug logs a few minutes after midnight.
influxdb continues POST /write ing logging every 5 seconds.
cronjob invoked at 00:07:01 causes InfluxDB to GET /ping and POST /query in response to a 
select last(... where Mac='xxx...  tz('EST')
influxdb continues POST /write ing until 
00:10:22
RuuviCollector reports 
... Apr 10, 2019 12:10:36 AM org.influxdb.impl.BatchProcessor write
...SEVERE: Batch could not be sent. Data will be lost
...org.influxdb.InfluxDBIOException: java.net.SocketTimeoutException: timeout
...  ... many java trace messages follow slowly 1 every second +/-

00:12:32  influxdb reports 5 additional POST /writes until
00:12:39 fatal error: runtime: out of memory, 
00:12:41 runtime stack:, 
00:12:48 runtime.throw(0xbe8b5d, 0x16)
00:12:51 ...panic.go:608 +0x5c, em_linux.go:156 +0xac, malloc.go:1230 +0x94,
malloc.go:525 +0x54, mheap.go:920 +0x2c, ... mcentral.go:232 +0x94, ...
asm_arm.s:354 +0x84, proc.go:1229...
00:12:52 runtime.systemstack_switch()...

00:12:52 RuuviCollector reports  ...SEVERE: Batch could not be sent.... again

influxdb continues with stack_switch, runtime.goexit(), main.main()
goroutine 5 [syscall, 7383 minutes]:
more,more
00:12:53 systemd[1]: Stopping RuuviCollector...
more,more
user.emerg pi:  ++ /home/pi/RuuviCollector got sig TERM
systemd[1]: ruuvicollector.service: Main process exited, code=exited, status=15/n/a

last messages for influxdb  (for a while!)
00:12:53 
#011/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:499 +0x40
created by github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).enableSnapshotCompactions
#011/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:499 +0xfc

systemd starts RuuviCollector immediately (even though RestartSec=300)
RuuviCollector begins a 240 second delay

 00:13:01
(vmstat 5 2 | logger -p kern.notice  )
 procs -----------memory---------- ---swap-- -----io---- -system-- ------cpu-----
  r  b   swpd   free   buff  cache   si   so    bi    bo   in   cs us sy id wa st
  3  0      0 290356   4640  75312    0    0    45    19   54   78 18  1 80  0  0 
  1  0      0 278452   4640  77488    0    0   435     0  252  306 98  2  0  0  0

00:15:13 suddenly influx fatal error: runtime: out of memory (no other messages since 00:12:53 !!)
influxdb crying continues
00:15:15 systemd Stopping RuuviCollector...
00:15:15 systemd[1]: Stopped InfluxDB 
influxdb crying continues
00:15:16 last message from influxdb:
 created by net/http.setRequestCancel
 client.go:320 +0x210
00:17:02 cronjob select to influxdb fails (no word from influxdb)

00:19:08 fatal error: runtime: out of memory,  and crying...

After 
00:19:10 systemd starts influxdb again
00:21:34 fatal error: runtime: out of memory,  and crying...
00:21:36 start, 

00:37:52 influxdb reports POST /write and happily continues reporting every 5 seconds
select last occurs at 07,17,27,37,47,57 
chronograf  starts a session at 10:30:07
writes continue every 5 seconds

Things I find interesting :
At first sign of a problem influxdb writes log messages very slowly then blasts out all the details 
7383 minutes is 123 hours I don't think so!

<h3>Additional information </h3>
System is a raspberry pi zero W, connected to network via wifi.

Linux pi93graf 4.14.98+ #1200 Tue Feb 12 20:11:02 GMT 2019 armv6l GNU/Linux

InfluxDB v1.7.4 (git: 1.7 ef77e72f435b71b1ad6da7d6a6a4c4a262439379)

config (none comments)
+++
wal-fsync-delay = "5s"
trace-logging-enabled = false
query-log-enabled = false
cache-max-memory-size = "300m"
series-id-set-cache-size = 100
query-timeout = "90s"
log-queries-after = "10s"
level = "warn"
+++

vm.swappiness = 10
vm.dirty_background_ratio = 5
vm.dirty_ratio = 80
tvservice --off  

/lib/systemd/system/ruuvicollector.service includes
RestartSec=300

/etc/systemd/system/ruuvicollector.service.requires/influxdb.service -> /lib/systemd/system/influxdb.service
 and RuuviCollector.sh includes a 240 second delay before starting java program

RuuviCollector.sh includes
trap 'logger --priority user.emerg " ++ /home/pi/RuuviCollector got sig TERM" ;exit 15' TERM

Complete log is at [http://MyBeacons.info/influx-memoryfail2.log.gz ](http://MyBeacons.info/influx-memoryfail2.log.gz )

I have just added RestartSec=240 to /etc/systemd/system/influxd.service

Any and all comments, suggestions are appreciated.

**Comment by DG12** [comment]
Again last night
Apr 11, 2019 12:10:25 AM org.influxdb.impl.BatchProcessor write

**Comment by DG12** [comment]
Again

BUT no "BAD" messages from influxdb only continuing 
POST /write...
POST /query on schedule, sucessfully retrieval: 
however 
00:23:06 , 00:23:11, 00:23:16, 00:23:26 , 00:23:31, 00:23:36 POST /write every 5 seconds as usual 
THEN 00:23:43 ! i.e. 2 seconds late!
THEN NO additional messages from influxdb 9 seconds later:
RuuviCollector Apr 12, 2019 12:23:54 AM  SEVERE: Batch could not be sent. Data will be lost... timeout
00:24:02 influxdb finally reports POST /write 27 seconds since previous message!
continues every 5 seconds
RuuviCollector ..12:27:24 AM .SEVERE: Batch could not be sent. Data will be lost... timeout
RuuviCollector . 12:27:41 AM SEVERE: Batch could not b...
RuuviCollector .12:27:59 AM SEVERE: Batch could not b...
00:27:14 a single influxdb POST/write
RuuviCollector .12:28:13 AM SEVERE: Batch could not b...
 00:28:11 a single influxdb POST /write
RuuviCollector 12:28:24 SEVERE: Batch could not b...
00:28:20 influxdb begins continually every 5 seconds 
+++. UNTIL
RuuviCollector  3:57:25 PM fails 5 times with no intervening influxdb messages
then 6 POST/Write at 15:58:39,15:58:40,15:57:47(sic),15:58:38 !, 15:58:39, 15:58:39 !
a query
a POST/write 15:58:48
 RuuviCollector . 3:58:58 PM SEVERE: Batch could not b...
Then influxdb continues 
++++++++++
atsar data at the time of failure
```
00:00:01  memtot memfree buffers   cached  slabmem      swptot swpfree  _mem_
00:21:16    433M     34M     87M      65M      87M       2047M   2047M
00:21:31    433M     27M     75M      63M      82M       2047M   2047M
00:21:46    433M     29M     46M      52M      54M       2047M   2047M
00:22:01    433M     40M     37M      46M      46M       2047M   2047M
00:22:16    433M     40M     37M      47M      46M       2047M   2047M
00:22:31    433M     40M     37M      47M      46M       2047M   2047M
00:22:46    433M     40M     37M      47M      46M       2047M   2047M
00:23:01    433M     40M     37M      47M      46M       2047M   2047M
00:23:16    433M     40M     37M      47M      46M       2047M   2047M
00:23:31    433M     25M     37M      52M      46M       2047M   2047M
00:23:46    433M     23M      0M       9M      12M       2047M   2042M
00:24:01    433M     24M      0M      11M      12M       2047M   2039M
00:24:16    433M     22M      0M      14M      11M       2047M   2038M
00:24:31    433M     23M      0M      13M      11M       2047M   2037M
00:24:46    433M     20M      0M      17M      11M       2047M   2037M
00:25:01    433M     24M      0M      12M      11M       2047M   2037M
00:25:16    433M     25M      0M      11M      11M       2047M   2037M
00:25:31    433M     24M      0M      12M      11M       2047M   2037M
00:25:46    433M     22M      0M      15M      11M       2047M   2037M
00:26:01    433M     23M      0M      14M      11M       2047M   2037M
00:26:16    433M     23M      0M      14M      11M       2047M   2037M
00:26:31    433M     24M      0M      13M      11M       2047M   2037M
00:26:46    433M     24M      0M      13M      11M       2047M   2037M
00:27:01    433M     22M      0M      15M      11M       2047M   2037M
00:27:16    433M     22M      0M      11M      11M       2047M   2035M
00:27:31    433M     24M      0M       9M      11M       2047M   2034M
00:27:46    433M     24M      0M      10M      11M       2047M   2034M
00:28:01    433M     22M      0M      11M      11M       2047M   2033M
00:28:16    433M     24M      0M       9M      11M       2047M   2033M
00:28:31    433M     24M      2M      14M      11M       2047M   2033M
00:28:46    433M     20M      0M      19M      11M       2047M   2033M
00:29:01    433M     24M      0M      16M      11M       2047M   2032M
00:29:16    433M     22M      1M      17M      11M       2047M   2032M
00:29:31    433M     25M      1M      15M      11M       2047M   2032M
00:30:01    433M     21M      2M      17M      11M       2047M   2032M
```


Note:
swap has been changed to a partition 
changed: vm.swappiness = 5 from 10; unchanged: vm.dirty_background_ratio = 5; vm.dirty_ratio = 80

**Comment by DG12** [comment]
This issue still occurs frequently. I have reduced the amount of data being inserted into the DB and set up  RETENTION POLICIES in a  effort to reduce the total database size.
name    duration  shardGroupDuration replicaN default
----    --------  ------------------ -------- -------
autogen 4320h0m0s 168h0m0s           1        true


I have also added a delay in the processing of the syslog files which I believe is causing some load on the system at midnight coincidently the time the problem occurs.

I am still hoping for any assistance, recommendations or requests for additional information.
Thank you.
